
I created this script to re-analyse the A375 libraries generated by Darcie before the screen with an aim to create scatterplots logFC vs. logt0 for t7 and t14.

The location of the data was presented in [20160616_processing.md](20160616_processing.md) and the way we want to process it is in [20161011_processing.md](20161011_processing.md)

I will take this opportunity to play with the `bowtie2` alignment options too.


# Processing of fastq files to count table

## Location of fastq files

The files are in `martin03@nas-srv001:/archive/Groups/SBLab/fs04/martin03/repository/fastq/` and the libraries prefix is `SLX-11622`

First, create receiving folders in `uk-cri-lcst01`:

```bash
cd /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data
mkdir 20170301
mkdir 20170301/fastq
```

Second, copy them from `nas-srv001` to `uk-cri-lcst01`,

```bash
rsync -arvuP /archive/Groups/SBLab/fs04/martin03/repository/fastq/SLX-11622*.s_1.r_1.fq.gz martin03@uk-cri-lcst01:/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/fastq
rsync -arvuP /archive/Groups/SBLab/fs04/martin03/repository/fastq/SLX-11622.HYLFFBGXX.s_1.barcodesummary.txt martin03@uk-cri-lcst01:/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/fastq
rsync -arvuP /archive/Groups/SBLab/fs04/martin03/repository/fastq/SLX-11622.HYLFFBGXX.s_1.contents.csv martin03@uk-cri-lcst01:/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/fastq

```

Looking at `SLX-11622.HYLFFBGXX.s_1.barcodesummary.txt` and comparing them with the previous barcodesummary files. Here we have:

* Total: 186562486. Lost: 9.29%


## Quality check

In uk-cri-lcst01, run an interactive session (`bsub` does not seem to allow me to submit jobs):

```bash
bsub -R "rusage[mem=8192]" -q interactive -Is bash
```

```bash
cd /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/fastq
mkdir ../fastqc

nohup /home/martin03/sw/fastqc/fastqc --noextract -q -o ../fastqc SLX-11622.RPI* &
rsync ../fastqc/*.html martin03@nm149s012925:/home/cri.camres.org/martin03/Desktop
rm nohup.out
```

In nm149s012925, having a look at the reports:

```bash
cd /home/cri.camres.org/martin03/Desktop
firefox *.html &
```

There is drop in quality of base 1 too.


## Trimming

```bash
cd /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/fastq
mkdir ../fastq_trim

for fq in SLX-11622.RPI*;
do
  nohup zcat $fq | fastx_trimmer -Q33 -l22 | pigz > ../fastq_trim/${fq%.fq.gz}.trim.fq.gz &
done
```


## Alignment using `bowtie2`

### Generate fasta file containing hairpins

The file was already generated before. It is here:

`/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20160420/Hs_WG_pool_data_P003_with_12Nic_3.fa`


### Generate index for hairpins fasta file using bowtie2-build

The indexes were already generated before and are available here:

`/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20160420/*.bt2`


### Align

```bash
cd /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/fastq_trim
IND="/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20160420/Hs_WG_pool_data_P003_with_12Nic_3"
mkdir ../sam/

for fq in SLX-11622.RPI*;
do
  nohup zcat $fq | bowtie2 --norc --no-hd -x $IND -U - -S ../sam/${fq%.trim.fq.gz}.sam &
done

cd ../sam
cat SLX-11622.RPI25.HYLFFBGXX.s_1.r_1.sam | cut -f2 | sort | uniq -c
#8554193 0
# 718164 4
cat SLX-11622.RPI26.HYLFFBGXX.s_1.r_1.sam | cut -f2 | sort | uniq -c
#6995413 0
# 514009 4
```

92-95% of reads with at least one reported alignment. OK.


### Align (with default parameters)

```bash
cd /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/fastq_trim
IND="/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20160420/Hs_WG_pool_data_P003_with_12Nic_3"
mkdir ../sam_default/

for fq in SLX-11622.RPI*;
do
  nohup zcat $fq | bowtie2 -x $IND -U - -S ../sam_default/${fq%.trim.fq.gz}.sam &
done

cd ../sam_default
samtools view SLX-11622.RPI25.HYLFFBGXX.s_1.r_1.sam | cut -f2 | sort | uniq -c
#8554193 0
# 718164 4
samtools view SLX-11622.RPI26.HYLFFBGXX.s_1.r_1.sam | cut -f2 | sort | uniq -c
#6995413 0
# 514009 4

```

92-95% of reads with at least one reported alignment. OK.

The results with and without `--norc --no-hd` are the same, which is encouraging. There are however several sequences (e.g. TTL__sherwood_1u__7_Pool12) which are potentially duplicated in the reference file

[W::sam_hdr_parse] duplicated sequence 'C3__sherwood__1__1_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C9__sherwood_1u__2__2_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F3__sherwood_1u__3__3_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F5__sherwood_1u__2__2_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C9__sherwood__1__1_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F11__sherwood__2__2_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F8__sherwood__1__1_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C9__sherwood_1u__3__3_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F8__sherwood_1u__1__1_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F10__sherwood_1u__3__3_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F10__sherwood_1u__2__2_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C3__sherwood_1u__4__4_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F5__sherwood__1__1_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F5__sherwood_1u__4__4_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F7__sherwood_1u__3__3_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C2__sherwood_1u__3__3_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C3__sherwood_1u__3__3_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F2__sherwood__2__2_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C2__sherwood_1u__1__1_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C2__sherwood__1__1_Pool1'
[W::sam_hdr_parse] duplicated sequence 'F2__sherwood_1u__3__3_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C3__sherwood__3__3_Pool1'
[W::sam_hdr_parse] duplicated sequence 'C6__sherwood_1u__3__3_Pool2'
[W::sam_hdr_parse] duplicated sequence 'DUX4__sherwood__1__1_Pool3'
[W::sam_hdr_parse] duplicated sequence 'TTL__sherwood_1u__1__1_Pool4'
[W::sam_hdr_parse] duplicated sequence 'DUX4__sherwood__3__3_Pool4'
[W::sam_hdr_parse] duplicated sequence 'TTL__sensor__1_Pool4'
[W::sam_hdr_parse] duplicated sequence 'GCOM1__1__1_Pool7'
[W::sam_hdr_parse] duplicated sequence 'KIAA0391__2__2_Pool7'
[W::sam_hdr_parse] duplicated sequence 'DUX4__1__1_Pool8'
[W::sam_hdr_parse] duplicated sequence 'TTL__3__3_Pool9'
[W::sam_hdr_parse] duplicated sequence 'DUX4__2__2_Pool10'
[W::sam_hdr_parse] duplicated sequence 'CMTM7__sensor__2_Pool10'
[W::sam_hdr_parse] duplicated sequence 'ZNF467__3__2_Pool10'
[W::sam_hdr_parse] duplicated sequence 'QTRTD1__sensor__2_Pool10'
[W::sam_hdr_parse] duplicated sequence 'CAV2__1__1_Pool10'
[W::sam_hdr_parse] duplicated sequence 'ZNF383__4__1_Pool10'
[W::sam_hdr_parse] duplicated sequence 'CHAT__sensor__2_Pool10'
[W::sam_hdr_parse] duplicated sequence 'CCR10__sensor__1_Pool10'
[W::sam_hdr_parse] duplicated sequence 'SSX4B__4__4_Pool10'
[W::sam_hdr_parse] duplicated sequence 'SSX4__4__4_Pool10'
[W::sam_hdr_parse] duplicated sequence 'KRTAP5-3__sensor__2_Pool10'
[W::sam_hdr_parse] duplicated sequence 'KRTAP5-8__sensor__1_Pool10'
[W::sam_hdr_parse] duplicated sequence 'DUX4__4__4_Pool11'
[W::sam_hdr_parse] duplicated sequence 'TTL__sherwood_1u__7_Pool12'


```bash
grep "TTL__sherwood_1u__7_Pool12" /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20160420/Hs_WG_pool_data_P003_with_12Nic_3.fa -B1
#>TTL__sherwood_1u__7_Pool12
#TAAGATTCAGGGAACCATGTGC
#>TTL__sherwood_1u__7_Pool12
#TAAGATTCAGGGAACCATGTGC
```

After grepping some of them, they are exactly the same sequence present twice, found the same in the original `.csv` file.

```bash
grep "TTL__sherwood_1u__7" /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20160420/Hs_WG_pool_data_P003_with_12Nic_3.csv
#ULTRA-3422068,TTL,150465,NM_153712,TTL__sherwood_1u__7,10,TGCTGTTGACAGTGAGCGACACATGGTTCCCTGAATCTTATAGTGAAGCCACAGATGTATAAGATTCAGGGAACCATGTGCTGCCTACTGCCTCGGA,204-0816_A10,Pool #12
#ULTRA-3482859,TTL,646982,NR_024505,TTL__sherwood_1u__7,10,TGCTGTTGACAGTGAGCGACACATGGTTCCCTGAATCTTATAGTGAAGCCACAGATGTATAAGATTCAGGGAACCATGTGCTGCCTACTGCCTCGGA,204-0816_B10,Pool #12
```

The sequences are also the same therefore these were just taken as one example really.



### Build table of counts

Generating counts:

```bash
cd /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/sam
mkdir ../counts

for sam in SLX-11622.RPI*.sam;
do
nohup cat $sam | cut -f3 | sort | uniq -c > ../counts/${sam%.sam}.counts &
done
```

Search one of the Pool8 sequences found duplicated above e.g. DUX4__1__1_Pool8, to check whether it is present in the counts files at all

```bash
cd ../counts
grep "DUX4__1__1_Pool8" *.counts
#SLX-11622.RPI25.HYLFFBGXX.s_1.r_1.counts:   1654 DUX4__1__1_Pool8
#SLX-11622.RPI26.HYLFFBGXX.s_1.r_1.counts:   1438 DUX4__1__1_Pool8
#SLX-11622.RPI27.HYLFFBGXX.s_1.r_1.counts:   2127 DUX4__1__1_Pool8
#SLX-11622.RPI28.HYLFFBGXX.s_1.r_1.counts:   1910 DUX4__1__1_Pool8
#SLX-11622.RPI29.HYLFFBGXX.s_1.r_1.counts:   1599 DUX4__1__1_Pool8
#SLX-11622.RPI30.HYLFFBGXX.s_1.r_1.counts:   1657 DUX4__1__1_Pool8
#SLX-11622.RPI31.HYLFFBGXX.s_1.r_1.counts:   1751 DUX4__1__1_Pool8
#SLX-11622.RPI32.HYLFFBGXX.s_1.r_1.counts:   1935 DUX4__1__1_Pool8
#SLX-11622.RPI33.HYLFFBGXX.s_1.r_1.counts:   2198 DUX4__1__1_Pool8
#SLX-11622.RPI34.HYLFFBGXX.s_1.r_1.counts:   2207 DUX4__1__1_Pool8
#SLX-11622.RPI35.HYLFFBGXX.s_1.r_1.counts:   2114 DUX4__1__1_Pool8
#SLX-11622.RPI36.HYLFFBGXX.s_1.r_1.counts:   1450 DUX4__1__1_Pool8
#SLX-11622.RPI37.HYLFFBGXX.s_1.r_1.counts:   1969 DUX4__1__1_Pool8
#SLX-11622.RPI38.HYLFFBGXX.s_1.r_1.counts:   1652 DUX4__1__1_Pool8
#SLX-11622.RPI39.HYLFFBGXX.s_1.r_1.counts:   2508 DUX4__1__1_Pool8
#SLX-11622.RPI40.HYLFFBGXX.s_1.r_1.counts:   2036 DUX4__1__1_Pool8
#SLX-11622.RPI41.HYLFFBGXX.s_1.r_1.counts:   1830 DUX4__1__1_Pool8
#SLX-11622.RPI42.HYLFFBGXX.s_1.r_1.counts:   1440 DUX4__1__1_Pool8
#SLX-11622.RPI43.HYLFFBGXX.s_1.r_1.counts:   1705 DUX4__1__1_Pool8
#SLX-11622.RPI44.HYLFFBGXX.s_1.r_1.counts:   2015 DUX4__1__1_Pool8
#SLX-11622.RPI45.HYLFFBGXX.s_1.r_1.counts:   1878 DUX4__1__1_Pool8

```

Good!

Now use python to parse the `.counts` files and put a table together:

```python
import os
import csv

libraries_dict = {}

with open('/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/fastq/SLX-11622.HYLFFBGXX.s_1.contents.csv', 'rb') as csvfile:
  reader = csv.reader(csvfile)
  header = reader.next()
  for row in reader:
    libraries_dict["%s.%s.HYLFFBGXX.s_1.r_1" % (row[0], row[1])] = row[3]


hp_lib = {}
hps = set()

for f in os.listdir("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/counts"):
  ifile = open("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/counts/%s" % f, "r")
  ilines = ifile.readlines()
  ifile.close()
  print f.replace(".counts", "")
  for l in ilines[1:]:
    fields = l.split()
    c = fields[0]
    hp = fields[1]
    hp_lib[(hp, f.replace(".counts", ""))] = c
    hps.add(hp)


len(hps) # 38817, number of unique hairpins aligned in all libraries, 100 * (38817 / 113002) = 34% of the total found in Hs_WG_pool_data_P003_with_12Nic_3.csv (see 113002 above)


ofile = open("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/20170301_counts.txt", "w")

ofile.write("key\t%s\n" % ("\t".join([libraries_dict[lib] for lib in sorted(libraries_dict)])))

for hp in sorted(hps):
  ofile.write("%s" % hp)
  for lib in sorted(libraries_dict):
    if (hp,lib) in hp_lib:
      ofile.write("\t%s" % hp_lib[(hp, lib)])
    else:
      ofile.write("\t0")
  ofile.write("\n")

ofile.close()

```


## Moving `.fastq` files to archive and remove unnecessary files

We do not need to move files to the `archive` because they were already taken from there in the first place.

Remove unnecessary files. In `uk-cri-lcst01`.

```bash
cd /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301
rm -rf fastq fastqc fastq_trim sam sam_default counts
```

We only keep `/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170301/20170301_counts.txt`
