
This script investigates whether genes that came up significant in the screen are actually enriched in COSMIC genes. It borrows bits from:

- [20160226_tsg_ocg.md](https://github.com/sblab-bioinformatics/projects/blob/master/20150501_methylation_brain/20160226_tsg_ocg.md#genes-for-which-mutations-have-been-causally-implicated-in-cancer--cosmic-forbes2015)
- [20170306_functional_analysis.md](20170306_functional_analysis.md)

As in 20170314, the COSMIC cancer gene [census](https://cancer.sanger.ac.uk/census) collects 616 genes for which mutations have been causally implicated in cancer.

I manually downloaded the data by clicking on TSV and copying the file over to lustre.

In `nm149s012925`,

```bash
cd /home/cri.camres.org/martin03/Desktop
mv Census_allTue\ Mar\ 14\ 09_57_14\ 2017.tsv Census_all_Tue_Mar14_09_57_14_2017.tsv
cat Census_all_Tue_Mar14_09_57_14_2017.tsv | cut -f3 | head # geneid is the third column
```

Other columns have information about the genomic location, the type of cancer where it is most commonly seen, type of mutation and if it is TSG (tumour suppressor gene)

In `uk-cri-lcst01`,

```bash
cd /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data
mkdir 20170314
cd 20170314
rsync -arvuP martin03@nm149s012925:/home/cri.camres.org/martin03/Desktop/Census_all_Tue_Mar14_09_57_14_2017.tsv .
```

```R
library(data.table)

# Adjust width
options(width=250)


# Load lists of sensitiser genes
pds_only <- fread("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170307/20170130_PDSonly_sensitiser_genes.txt")
phendc3_only <- fread("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170307/20170130_PhenDC3only_sensitiser_genes.txt")
pds_phendc3_only <- fread("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170307/20170130_PDSandPhenDC3only_sensitiser_genes.txt")
dmso <- fread("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170307/20170130_DMSO_sensitiser_genes.txt")

nrow(pds_only) # 224
nrow(phendc3_only) # 512
nrow(pds_phendc3_only) # 107
nrow(dmso) # 408


# Load mapping between gene names and gene ids
genename_geneid <- fread("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20160420/Hs_WG_pool_data_P003_with_12Nic_3.gene_name.gene_id.txt")
setnames(genename_geneid, c("gene_name","gene_id"))


# Merge
pds_only <- merge(pds_only, genename_geneid, by.x = 'gene', by.y='gene_name', all.x=TRUE)
sum(is.na(pds_only$gene_id)) # 2
pds_only[is.na(pds_only$gene_id)] # MARCH3 MARCH6
pds_only[gene=="MARCH3", gene_id := 115123] # http://www.uniprot.org/uniprot/Q86UD3
pds_only[gene=="MARCH6", gene_id := 10299] # http://www.uniprot.org/uniprot/O60337
sum(is.na(pds_only$gene_id)) # 0
pds_only_geneids <- unique(pds_only$gene_id)

phendc3_only <- merge(phendc3_only, genename_geneid, by.x='gene', by.y='gene_name', all.x=TRUE)
sum(is.na(phendc3_only$gene_id)) # 0
phendc3_only_geneids <- unique(phendc3_only$gene_id)

pds_phendc3_only <- merge(pds_phendc3_only, genename_geneid, by.x='gene', by.y='gene_name', all.x=TRUE)
sum(is.na(pds_phendc3_only$gene_id)) # 0
pds_phendc3_only_geneids <- unique(pds_phendc3_only$gene_id)

dmso <- merge(dmso, genename_geneid, by.x='gene', by.y='gene_name', all.x=TRUE)
sum(is.na(dmso$gene_id)) # 1
dmso[is.na(dmso$gene_id)] # MARCH9
dmso[gene=="MARCH9", gene_id := 92979] # http://www.uniprot.org/uniprot/Q86YJ5
sum(is.na(dmso$gene_id)) # 0
dmso_geneids <- unique(dmso$gene_id)


# Load COSMIC table
cosmic <- fread("/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/data/20170314/Census_all_Tue_Mar14_09_57_14_2017.tsv")
cosmic[,"Entrez GeneId", with = FALSE] # the header names contain spaces, therefore we need to use an alternative approach https://rawgit.com/wiki/Rdatatable/data.table/vignettes/datatable-intro.html
length(data.frame(cosmic)[,3]) # gene ids is the 3rd column, 616 gene ids
length(unique(data.frame(cosmic)[,3])) # 615 unique gene ids
cosmic_geneids <- unique(data.frame(cosmic)[,3])


# Counts and fisher.test()
## pds_only
cnt_pds_only_cosmic <- length(intersect(pds_only_geneids, cosmic_geneids))
cnt_pds_only_NOcosmic <- length(setdiff(pds_only_geneids, cosmic_geneids))
cnt_NOpds_only_cosmic <- length(setdiff(cosmic_geneids, pds_only_geneids))
cnt_NOpds_only_NOcosmic <- length(setdiff(genename_geneid$gene_id, union(cosmic_geneids, pds_only_geneids)))

table_pds_only <- matrix(c(cnt_pds_only_cosmic, cnt_pds_only_NOcosmic, cnt_NOpds_only_cosmic, cnt_NOpds_only_NOcosmic), nrow = 2)

fisher.test(table_pds_only) # p-value = 0.0001347, there is a significant enrichment of pds_only sensitisers in COSMIC genes

(cnt_pds_only_cosmic/cnt_pds_only_NOcosmic)/(cnt_NOpds_only_cosmic/cnt_NOpds_only_NOcosmic) # 2.821697 fold enrichment

write.table(pds_only[gene_id %in% intersect(pds_only_geneids, cosmic_geneids)], file = "/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_PDSonly_sensitiser_genes_cosmic.txt", quote = FALSE, sep = "\t", row.names=FALSE)
system("rsync --remove-source-files /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_PDSonly_sensitiser_genes_cosmic.txt martin03@nm149s012925:/media/itshare/research/sblab/group/public_folders/martin03/20160418_shRNAscreen_katie_darcie/20170314/tables")


## phendc3_only
cnt_phendc3_only_cosmic <- length(intersect(phendc3_only_geneids, cosmic_geneids))
cnt_phendc3_only_NOcosmic <- length(setdiff(phendc3_only_geneids, cosmic_geneids))
cnt_NOphendc3_only_cosmic <- length(setdiff(cosmic_geneids, phendc3_only_geneids))
cnt_NOphendc3_only_NOcosmic <- length(setdiff(genename_geneid$gene_id, union(cosmic_geneids, phendc3_only_geneids)))

table_phendc3_only <- matrix(c(cnt_phendc3_only_cosmic, cnt_phendc3_only_NOcosmic, cnt_NOphendc3_only_cosmic, cnt_NOphendc3_only_NOcosmic), nrow = 2)

fisher.test(table_phendc3_only) # p-value = 0.1267, there is enrichment of phendc3_only sensitisers in COSMIC genes but it is not significant

(cnt_phendc3_only_cosmic/cnt_phendc3_only_NOcosmic)/(cnt_NOphendc3_only_cosmic/cnt_NOphendc3_only_NOcosmic) # 1.419067 fold enrichment (not significant)

write.table(phendc3_only[gene_id %in% intersect(phendc3_only_geneids, cosmic_geneids)], file = "/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_PhenDC3only_sensitiser_genes_cosmic.txt", quote = FALSE, sep = "\t", row.names=FALSE)
system("rsync --remove-source-files /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_PhenDC3only_sensitiser_genes_cosmic.txt martin03@nm149s012925:/media/itshare/research/sblab/group/public_folders/martin03/20160418_shRNAscreen_katie_darcie/20170314/tables")


## pds_phendc3_only
cnt_pds_phendc3_only_cosmic <- length(intersect(pds_phendc3_only_geneids, cosmic_geneids))
cnt_pds_phendc3_only_NOcosmic <- length(setdiff(pds_phendc3_only_geneids, cosmic_geneids))
cnt_NOpds_phendc3_only_cosmic <- length(setdiff(cosmic_geneids, pds_phendc3_only_geneids))
cnt_NOpds_phendc3_only_NOcosmic <- length(setdiff(genename_geneid$gene_id, union(cosmic_geneids, pds_phendc3_only_geneids)))

table_pds_phendc3_only <- matrix(c(cnt_pds_phendc3_only_cosmic, cnt_pds_phendc3_only_NOcosmic, cnt_NOpds_phendc3_only_cosmic, cnt_NOpds_phendc3_only_NOcosmic), nrow = 2)

fisher.test(table_pds_phendc3_only) # p-value = 0.002501, there is a significant enrichment of pds_phendc3_only sensitisers in COSMIC genes

(cnt_pds_phendc3_only_cosmic/cnt_pds_phendc3_only_NOcosmic)/(cnt_NOpds_phendc3_only_cosmic/cnt_NOpds_phendc3_only_NOcosmic) # 3.110335 fold enrichment

write.table(pds_phendc3_only[gene_id %in% intersect(pds_phendc3_only_geneids, cosmic_geneids)], file = "/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_PDSandPhenDC3only_sensitiser_genes_cosmic.txt", quote = FALSE, sep = "\t", row.names=FALSE)
system("rsync --remove-source-files /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_PDSandPhenDC3only_sensitiser_genes_cosmic.txt martin03@nm149s012925:/media/itshare/research/sblab/group/public_folders/martin03/20160418_shRNAscreen_katie_darcie/20170314/tables")


## sensitiser
sensitiser_geneids <- unique(c(pds_only$gene_id, phendc3_only$gene_id, pds_phendc3_only$gene_id))
length(sensitiser_geneids) # 843

cnt_sensitiser_cosmic <- length(intersect(sensitiser_geneids, cosmic_geneids))
cnt_sensitiser_NOcosmic <- length(setdiff(sensitiser_geneids, cosmic_geneids))
cnt_NOsensitiser_cosmic <- length(setdiff(cosmic_geneids, sensitiser_geneids))
cnt_NOsensitiser_NOcosmic <- length(setdiff(genename_geneid$gene_id, union(cosmic_geneids, sensitiser_geneids)))

table_sensitiser <- matrix(c(cnt_sensitiser_cosmic, cnt_sensitiser_NOcosmic, cnt_NOsensitiser_cosmic, cnt_NOsensitiser_NOcosmic), nrow = 2)

fisher.test(table_sensitiser) # p-value = 9.104e-06, there is significant enrichment of sensitisers in COSMIC genes

(cnt_sensitiser_cosmic/cnt_sensitiser_NOcosmic)/(cnt_NOsensitiser_cosmic/cnt_NOsensitiser_NOcosmic) # 2.050304 fold enrichment

write.table(genename_geneid[gene_id %in% intersect(sensitiser_geneids, cosmic_geneids)], file = "/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_sensitiser_genes_cosmic.txt", quote = FALSE, sep = "\t", row.names=FALSE)
system("rsync --remove-source-files /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_sensitiser_genes_cosmic.txt martin03@nm149s012925:/media/itshare/research/sblab/group/public_folders/martin03/20160418_shRNAscreen_katie_darcie/20170314/tables")


## dmso
cnt_dmso_cosmic <- length(intersect(dmso_geneids, cosmic_geneids))
cnt_dmso_NOcosmic <- length(setdiff(dmso_geneids, cosmic_geneids))
cnt_NOdmso_cosmic <- length(setdiff(cosmic_geneids, dmso_geneids))
cnt_NOdmso_NOcosmic <- length(setdiff(genename_geneid$gene_id, union(cosmic_geneids, dmso_geneids)))

table_dmso <- matrix(c(cnt_dmso_cosmic, cnt_dmso_NOcosmic, cnt_NOdmso_cosmic, cnt_NOdmso_NOcosmic), nrow = 2)

fisher.test(table_dmso) # p-value = 4.718e-05, there is significant enrichment of dmso sensitisers in COSMIC genes

(cnt_dmso_cosmic/cnt_dmso_NOcosmic)/(cnt_NOdmso_cosmic/cnt_NOdmso_NOcosmic) # 2.438204 fold enrichment

write.table(dmso[gene_id %in% intersect(dmso_geneids, cosmic_geneids)], file = "/lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_DMSO_sensitiser_genes_cosmic.txt", quote = FALSE, sep = "\t", row.names=FALSE)
system("rsync --remove-source-files /lustre/sblab/martin03/repository/20160418_shRNAscreen_katie_darcie/tables/20170314_DMSO_sensitiser_genes_cosmic.txt martin03@nm149s012925:/media/itshare/research/sblab/group/public_folders/martin03/20160418_shRNAscreen_katie_darcie/20170314/tables")

```
